oak_containers_hello_world_trusted_app/src/orchestrator_client.rs

```diff
@@ -86,3 +91,20 @@ impl OrchestratorClient {
         Ok(())
     }
 }
+
+impl RecipientContextGenerator for OrchestratorClient {
+    fn generate_recipient_context(
+        &self,
+        serialized_encapsulated_public_key: &[u8],
+    ) -> anyhow::Result<RecipientContext> {
+        let context =
+            executor::block_on(self.get_crypto_context(serialized_encapsulated_public_key))
```

Comment:
Blocking the current thread while waiting for the RPC call will be a performance problem in a production system.

I think this requires a bit of a redesign, but that can be done as a separate task. Please create an issue and add a TODO.

I think a possible design is to change the ServerEncryptor so that is can either be created with a RecipientContextGenerator or a RecipientContext. The async code to get the context can then be called from the async service implementation and the crypto part can remain completely free of async code.

Reply:
Created an issue and added a TODO: https://github.com/project-oak/oak/issues/4258

Reply:
Done in https://github.com/project-oak/oak/pull/4281